### DISCUSSION TITLE
Limiting number of reflections

### DISCUSSION THREAD

**Discussion by jumpjack** [ricktu288/ray-optics#157] [Ideas]
Do you think you could add such an option? [This simulation ](https://phydemo.app/ray-optics/simulator/#XQAAAAISAgAAAAAAAABEKcrGU8hqLFnpmU9EnTFrVnUsuHiP9Y34lelaIGbrNVQFbkuPLdhijfW256Z23BLnSjfifhyGMClTKdTbeNA4zSvsxiov8vCUNlj-LSRiZr4u2f0PBBfVuYzCBWn4k2uhno8KgYEKi5JHiy-HyL81Svo-9Ht0hapkG8ZiMkiWOWVZm-1JGdrTcoyhDBx-CgRw_v6QnX04ZDo1pLLALHH_0swBnhojsnrIUrvSg96lV1lMxmCRaRZNxbTYGClKtAHhtW0yPrSgoLY1k3jrEGG-DbdjE3GbeQ4OyZ8pvlr2aaekuEHkn0UcqeCw5LL1mj5fifqQt-OsOhoi1a6GpR_PkNt1G3kTGZDq_z_WCIc0uasf4KybFcwR8Oyd4dQ5x9JvaeEWbXholSxVVHp20-YpbsPx35mWnyDRuQxJoFpYJu2s8_pQcJEBxr_NaMvc1Qn9HtZbxq4HbwT0cYjGNOewAt2hPxcHB-abcNrlgl13--b4rHg7EQOkLe2DjQ1SEjRbz0su4P5a2ukcbGBv_YoBIA) looks really messy, but it consists of just ONE single ray! Although it's physically correct to show all the reflections with their attuenations, I think the simulation would be much more readable if the user could tune how many reflections should be visible: infinite, or a finite number.

A further thing which would make it more readable would be arrows showing where the light come from and goes to.

**Comment by ricktu288** [comment]
I think the reason it looks messy is not because of the attuenated rays, but because the backward and forward extension of each interaction are shown. So the best way to solve this is possibly to allow selecting which extensions should be shown.

**Comment by ricktu288** [comment]
For the second point, I've added a new option "Show Ray Arrows"

**Comment by MaxwellMayman** [comment]
For what its worth, I also could really use the ability to limit reflections.

**Reply by ricktu288** [reply]
A workaround is to use Tools -> Other -> More tools -> Custom Surface to make a mirror with finite reflectivity.

**Reply by ricktu288** [reply]
A "maximum ray depth" option has been added to Settings -> More scene settings.

**Comment by BlueCl0wn** [comment]
I have a similar albeit not the same problem. See in [this example](https://phydemo.app/ray-optics/simulator/#XQAAAAL0AwAAAAAAAABFqcrGU8hqLFnpmU9ERVKNOweovUDEmiq-K4rZmTs9BEk92L9QSAdjsxRoMWmtUqNNXBWI8xwKv-i6PyZpvMbwQY0-omblWbC99_5iYmub5gB0m58dk5G6PDweMDfFMCFLqIupV--yrF9AiVvubfaqLefEWTs9WKQngm4L4zlkh5dvccePUzfrgfXVPjoWJFd0-i--NJN2sikX8xWbCNrsB2nyUsoZ1YNMOX4LN2XMtD6LoYaMP1tH0a5w8u9MrlM0-9lyQAR7BIAJtwCcPBZjkBjA_L4h_oWMBN8qKaDVb709-w6iwxOBJK9jDFFxS_62Y1f5CM4Kaec1WvxHVoiII6gxxOr6_JTwr8njr89Iqp1vpAMnCoq8ODPmV_noJbxDSbOdsZT93g1Rhz75pT4312LeJOANqRjf4WIoe7OqJBJD0M_QszL7XbqpCJfxsDBGBqHU2-0A49Fbf1y19qR5ZnadhI6m7A0YbnmDKDaAP2M5DsxdcMh9dFiR4Aud2EiU1-Hi9Gp706dYv8vQk3WXmtQnOlxV7JvHnrv235kiCN4cJSkSoIvOYp8Pz_Y-zzyhCpO3cICoVd58CKBiBmg0ITj7nrX1No6JenXdly438tiXyE5tjb_HkSdIAv7y4qMTJWmUSi4SxtZqDoopQY1oxk0BzDZkEJ1mvJKfROnegU9KuQnxG7oJoDQ6zHLhMP43N83TGAAFMi4r6DXhpE6BM0njE9Lf1hQitG3TaWNewDdr8kOxGBJqjWnHSBe9KdxsuUk5fFMjZDlFxyYu3k3nEbvqu87Hum9GWjO-6C4b_DywLiRtlwj_9guDaA) the bottom setup. The back reflections at the lens surfaces are distracting in my case and an option to turn them off would be nice.

Regarding the arrows: I do like that feature. Something that would make them even more interesting is, if they would show the wavefront of a beam. This is probably a bit difficult to implement and only useful to few people, but there are definitely scenarios in which that feature would make sense.

**Reply by ricktu288** [reply]
For turning off reflections, I will consider adding a new options for glasses. But for now you may use Tools -> Other -> More tools -> Custom Surface and enter the Snell law to achieve that. Also see Tools -> Other -> Import modules -> Coated lens.

For the wavefront, that is indeed difficult to implement for now but the same difficulty also occurs also for #37, #137, and a more efficient implementation of #109 (see them for more information), so it is overall worth implementing.

**Reply by ricktu288** [reply]
A new option has been added in the "More option..." of glasses.

**Comment by ricktu288** [comment]
A "maximum ray depth" option has been added to Settings -> More scene settings.
